binding/Binding/SKPngChunkReader.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.ComponentModel;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace SkiaSharp
+{
+	/// <summary>
+	/// Base class for optional callbacks to retrieve meta/chunk data out of a PNG
+	/// <br></br> encoded image as it is being decoded.
+	/// <br></br> Used by SkCodec.
+	/// </summary>
+	public unsafe abstract class SKPngChunkReader : SKObject, ISKSkipObjectRegistration
+	{
+		private static readonly SKManagedPngChunkReaderDelegates delegates;
+		private readonly IntPtr userData;
+		private int fromNative;
+
+		static SKPngChunkReader()
+		{
+			delegates = new SKManagedPngChunkReaderDelegates
+			{
+				fReadChunk = ReadChunkInternal,
+				fDestroy = DestroyInternal,
+			};
+
+			SkiaApi.sk_managed_png_chunk_reader_set_procs(delegates);
+		}
+
+		protected SKPngChunkReader()
+			: base(IntPtr.Zero, true)
+		{
+			userData = DelegateProxies.CreateUserData(this, true);
+			Handle = SkiaApi.sk_managed_png_chunk_reader_new((void*)userData);
+
+			if (Handle == IntPtr.Zero)
+				throw new InvalidOperationException("Unable to create a new SKManagedAllocator instance.");
+		}
+
+		protected override void Dispose(bool disposing) =>
+			base.Dispose(disposing);
```

Comment:
This is not needed. It exists in other places because I have a dumb API checker that thinks overrides are "breaking".

Reply:
alright